devito/propagator.py

```diff
@@ -3,14 +3,20 @@
 import numpy as np
 from sympy import symbols, IndexedBase, Indexed
 from function_manager import FunctionDescriptor
-import os
 
 
 class Propagator(object):
-    _ENV_VAR_PROFILE = "DEVITO_PROFILE"
-
-    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False):
+    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False, cache_blocking=False, block_size=5):
         self.t = symbols("t")
+        self.cache_blocking = cache_blocking
+        try:
```

Comment:
I don't think `try`/`except` is appropriate here. Why not check for list/tuples via `if isinstance(Iterable):`, with `from collections import Iterable`?